src/middlewared/middlewared/plugins/disk_/utils.py

```diff
@@ -0,0 +1,26 @@
+def valid_zfs_partition_uuids():
+    # https://salsa.debian.org/debian/gdisk/blob/master/parttypes.cc for valid zfs types
+    # 516e7cba was being used by freebsd and 6a898cc3 is being used by linux
+    return (
+        '6a898cc3-1dd2-11b2-99a6-080020736631',
+        '516e7cba-6ecf-11d6-8ff8-00022d09712b',
+    )
+
+
+def dev_to_ident(name, sys_disks, uuids):
+    """Map a disk device (i.e. sda5) to its respective "identifier"
+    (i.e. "{serial_lunid}AAAA_012345")"""
+    try:
+        dev = sys_disks[name]
+    except KeyError:
+        return ''
```

Comment:
Presumably we shouldn't ever hit this return.  Log something?

Reply:
On the contrary. Many systems in the wild will run esoteric hardware or import disks/zpools from foreign systems that can (and often do) cause all kinds of "fun". 😄